### DISCUSSION TITLE
Is there a VipsForeignSavePDFTarget ? How to save an image as PDF?

### DISCUSSION THREAD

**Discussion by ademmler** [libvips/libvips#3551] [Q&A]
hi there,

can I save an image as a one page PDF? 
Is there a VipsForeignSavePDFTarget() ?

thx Alex

**Comment by jcupitt** [comment]
Hi, sorry, PDF save is not supported directly.

You can save as PDF via imagemagick, for example:

```
$ vips magicksave k2.jpg x.pdf
$ file x.pdf
x.pdf: PDF document, version 1.3, 1 pages
```

ImageMagick only supports file and buffer write, so `magicksave_target` isn't possible, sadly. You can write to memory though:

```
$ vips magicksave_buffer k2.jpg --format pdf
VIPS_TYPE_BLOB, data = 0x7ff20f76b010, length = 8994113
```

**Comment by ademmler** [comment]
Ok - thank your for explanation.
